### ISSUE TITLE
\ref{ and \cite{ autocompletion gobbles text in 3.10.x update

### ISSUE THREAD

**Issue by DrBones** [SublimeText/LaTeXTools#800]
Maybe specific to my current setup but it did not happen with version 3.9.3

If I place a \ref{ or \cite{ and use the autocompletion dialog on a line that has a subscript, i.e A_{test} the whole text after the subscript is gobbled up and replaced by the reference or citation. This happens regardless if the text is in $$, \\[\\] or just plain text. This also does not happen if only one label is available

MWE:
```
   \label{eq:testlabel1}
   \label{eq:testlabel2}
   In this line has no subscript math \ref{eq:testlabel1}
   This line $A_{hallo}$ contains subscript math \ref{
```
if I select a label to be referenced the last line becomes:
```
   This line A_f{eq:testlabel1}
```

For now, manually installing version 3.9.3 solved the issue.
If I can do anything to help debug this let me know but I am fairly new to Sublimetext plugins :)

By the way, awesome plugin you have made there. I would not know how to write my theses without it!

**Comment by bagasabisena** [comment]
Hi, I experience the same situation like it was mentioned above after the update to 3.10.x.

If a write the following sentence

`Reference A \cite{citekey_a} and reference B \cite{`

the sentence becomes

`Reference A \cite{citekey_ate{}`

It happens if there is a \cite command before another \cite command that I want the autocomplete on within a single line. The second command somehow will get merged to the previous command.

Please let me know if you need more diagnostics.

**Comment by DrBones** [comment]
Good to know I am not the only one and my setup is to blame :).

Your example works (gobbles) for me too. I deleted some more of my tex file and it actually happens as soon as there is **any** underscore "_" on the line where a ref or cite is to be placed.

**Comment by ig0774** [comment]
Yes. There was a piece of code that's trying to handle old-style `\ref_prefix` and `\cite_prefix` completions that was a bit too greedy. This has been fixed in v3.10.3, which I've just released. Hopefully it'll be on PackageControl soon. Thanks for the detailed bug report and sorry about any frustration this has caused!

**Comment by DrBones** [comment]
Amazing!

Thank you for your swift reply and fix. I have just updated to v3.10.3 and everything is well now. This has caused no frustration, only slight confusion and a nice excuse to procrastinate :).

I am closing this bug now
